### PR TITLE
Update Autorest C#

### PR DESCRIPTION
**Pull request by pakrym**
Parse enums case-insensitively

### CODE REVIEW

**Comment by bleroy** [comment]
Can you explain how you caused that change in generated code?

**Comment by pakrym** [comment]
That's the result of recent generator update https://github.com/Azure/autorest.csharp/pull/532

**Comment by tg-msft** [comment]
(And it only applies to the new Track 2 `Azure.Search` client library.)

**Review by tg-msft — APPROVED** [review]

File: sdk/search/Azure.Search/src/Generated/Models/AutocompleteMode.Serialization.cs [hunk 1]
```diff
@@ -19,12 +19,12 @@ internal static class AutocompleteModeExtensions
             _ => throw new ArgumentOutOfRangeException(nameof(value), value, "Unknown AutocompleteMode value.")
         };
 
-        public static AutocompleteMode ToAutocompleteMode(this string value) => value switch
+        public static AutocompleteMode ToAutocompleteMode(this string value)
```
Comment (heaths) [inline sdk/search/Azure.Search/src/Generated/Models/AutocompleteMode.Serialization.cs]: Do these changes also affect modelAsString enums / extensible enums?

**Review by heaths — COMMENTED** [review]
Overall, I have concerns. Other languages like TypeScript that generally are using a `union` send the string as-is, which means the server gets the casing of whatever the user supplied (which, for TS - but not JS - is compiler-validated. That may generate an error, but here we are case-insensitive? It's an inconsistent experience. Are we okay with that?

**Review by bleroy — COMMENTED** [review]
Thanks!

**Comment by pakrym** [comment]
> Overall, I have concerns. Other languages like TypeScript that generally are using a union send the string as-is, which means the server gets the casing of whatever the user supplied (which, for TS - but not JS - is compiler-validated. That may generate an error, but here we are case-insensitive? It's an inconsistent experience. Are we okay with that?

We are sending the casing that's defined in swagger but are being permissive (against the spec) to what we received. I've talked to python folks and they had to do the same because services are very inconsistent in what they are returning.

File: sdk/search/Azure.Search/src/Generated/Models/AutocompleteMode.Serialization.cs (hunk 1 above)
Comment (pakrym) [reply sdk/search/Azure.Search/src/Generated/Models/AutocompleteMode.Serialization.cs]: Filed https://github.com/Azure/autorest.csharp/issues/533

File: eng/CodeGeneration.targets [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <_AutoRestVersion>https://github.com/Azure/autorest/releases/download/autorest-3.0.6191/autorest-3.0.6191.tgz</_AutoRestVersion>
     <_AutoRestCoreVersion>3.0.6237</_AutoRestCoreVersion>
-    <_AutoRestCSharpVersion>https://github.com/Azure/autorest.csharp/releases/download/1.0.0-dev.20200311.4/autorest-csharp-v3-1.0.0-dev.20200311.4.tgz</_AutoRestCSharpVersion>
+    <_AutoRestCSharpVersion>https://github.com/Azure/autorest.csharp/releases/download/1.0.0-dev.20200312.2/autorest-csharp-v3-1.0.0-dev.20200312.2.tgz</_AutoRestCSharpVersion>
```
Comment (weshaggard) [inline eng/CodeGeneration.targets]: Are we at a place were we want to have PR's automatically submitted for these updates?

Comment (pakrym) [reply eng/CodeGeneration.targets]: I would love it if something can update the version and run the generator on everything.

Comment (weshaggard) [reply eng/CodeGeneration.targets]: I'm happy to help you put together a pipeline that can do this. Lets sync up at some point in the coming weeks to see how we can do this.

**Pull request merged**